File: 0985-sum-of-even-numbers-after-queries/0985-sum-of-even-numbers-after-queries.py
class Solution(object):
    def sumEvenAfterQueries(self, nums, queries):
        """
        :type nums: List[int]
        :type queries: List[List[int]]
        :rtype: List[int]
        """
        S = sum(num for num in nums if num % 2 == 0)
        answer = []
        
        for val,index in queries:
            if nums[index] % 2 == 0: S -= nums[index]
            nums[index] += val
            if nums[index] % 2 == 0: S += nums[index]
            answer.append(S)
            
        return answer

File: 0985-sum-of-even-numbers-after-queries/0985-sum-of-even-numbers-after-queries.cpp
class Solution {
public:
    vector<int> sumEvenAfterQueries(vector<int>& nums, vector<vector<int>>& queries) {
        vector<int> answer;
        for (int i = 0; i < queries.size(); i++ ){
            int even_sum = 0;
            // nums[index] = nums[index] + val
            nums[queries[i][1]] += queries[i][0];
            for (int j = 0; j < nums.size(); j++){
                if (nums[j] % 2 == 0){
                    even_sum += nums[j];
                }
            }
            answer.push_back(even_sum);
        }
        return answer;
    }
};